src/app/auth/pages/connection-page/connection-page.component.ts

```diff
@@ -13,25 +12,31 @@ import {LoginRequestPayload} from "../../models/login-request-payload";
 })
 export class ConnectionPageComponent implements OnInit {
 
-  title: string = 'Connexion';
-  user: User = {
+  title = 'Connexion';
+  user = {
     username: '',
     email: '',
     password: ''
   };
 
-  activeMsg: boolean = false;
-  msgClass: string = '';
-  msgText: string = '';
+  activeMsg = false;
+  msgClass = '';
+  msgText = '';
+  testeur = '';
 
   constructor(
+    private activatedRoute: ActivatedRoute,
     private userService: UserService,
     private authService: AuthService,
     private router: Router
   ) {
   }
 
   ngOnInit(): void {
+    this.activatedRoute.paramMap.subscribe(params => {
```

Comment:
```suggestion
    this.activatedRoute.queryParamMap.subscribe(params => {
```
C'est pas très clair malheureusement mais il faut utiliser `queryParamMap` et non `paramMap`
paramMap => /route/:param
queryParamMap => /route?param

Reply:
Avec queryParamMap, ça marche mais pour le fait le subscribe s'active que si le queryParams 'error' existe, c'est une autre histoire.
J'essaye : 
`ngOnInit(): void {
    this.activatedRoute.queryParamMap.subscribe(params => {
      const receivedError = params.get('error');
      console.log(receivedError);

      if (receivedError) {
        this.activatedRoute.queryParamMap.subscribe(params => {
          this.testeur = params.get('test') as string;
          console.log(this.testeur);
        });
      }
    });
  }`
mais ça marche pas bien.

Reply:
Creuse du côté des pipes

Reply:
J'ai trouvé que | c'est angular, et pipe() c'est RxJS. Cela ne me sert pas à grand chose mais c'est toujours ça.

Reply:
Je vois ce qu'il faut faire mais je ne sais pas du tout comment procéder.
Avec un if c'est clair, mais avec une fonction qui prends une fonction flêchée pour faire ce que fait le if, je vois pas comment. :(

Reply:
`ngOnInit(): void {
    this.activatedRoute.queryParamMap.pipe(
      // A mettre dans un filter()
      if (params.get('error') as string != undefined) {
        // alors tu peux passer
      }
    ).subscribe(params => {
      this.testeur = params.get('test') as string;
      console.log(this.testeur);
    });
  }`

Reply:
tu as 2 choses à savoir : 
* il existe l'operateur `filter()` qui permet de filtrer seulement les événements qui satisfont un prédicat (en gros une condition)
* `params` c'est une map. Les maps disposent de la méthode `.has(key: string)` qui renvoie `true` si la clé existe.